apps/src/lib/ui/SchoolInfoConfirmationDialog.jsx

```diff
@@ -15,80 +15,102 @@ export const styles = {
 class SchoolInfoConfirmationDialog extends Component {
   static propTypes = {
     schoolName: PropTypes.string,
-    onUpdate: PropTypes.func,
-    onConfirm: PropTypes.func,
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        id: PropTypes.number,
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        full_address: PropTypes.string
+      }).isRequired
+    }).isRequired,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.state = {
       showSchoolInterstitial: false,
-      schoolName: props.schoolName,
-      isOpen: true
+      schoolName: props.scriptData.existingSchoolInfo.school_name,
+      isOpen: props.isOpen || true
     };
   }
 
-  static defaultProps = {
-    onUpdate: () => {},
-    schoolName: ''
+  handleClickYes = () => {
+    const {authTokenName, authTokenValue} = this.props.scriptData;
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_last_confirmation_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => this.props.onClose())
+      .catch(error => {
+        this.setState({error});
```

Comment:
It looks like we never use this error state.  Should the component respond to this somehow?